### PR TITLE
Added unit tests, added target for running unit tests with coverage, did a few minor refactors along the way

### PR DESCRIPTION
**Pull request by sathiish-kumar**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/ExternalObjectReplicator/tests/test_copy_util.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import boto3
 import datetime
 from unittest.mock import patch, MagicMock
-from unittest import mock
+from unittest import mock, IsolatedAsyncioTestCase
```
Comment (DevanathanSabapathy1) [inline tools/ExternalObjectReplicator/tests/test_copy_util.py:5]: I am not able to import IsolatedAsyncioTestCase in python 3.7.1
Using this version so sync manager works. ( I assume you run a later version of python and I hope sync manager still works with the newer versions.)

File: common/tests/test_aws_service.py [hunk 2]
```diff
@@ -1,22 +1,53 @@
+import base64
 import datetime
-from unittest import TestCase, mock
+from unittest import TestCase, mock, IsolatedAsyncioTestCase
```
Comment (DevanathanSabapathy1) [inline common/tests/test_aws_service.py:3]: Same reason as below comment in test_copy_util.py , I believe IsolatedAsyncioTestCase is available in the later versions.

**Review by DevanathanSabapathy1 — APPROVED** [review]

**Review by jiezhen-chen — APPROVED** [review]

**Review by karle-nishi — APPROVED** [review]

File: tools/ExternalObjectReplicator/tests/test_copy_util.py (hunk 1 above)
Comment (karle-nishi) [reply tools/ExternalObjectReplicator/tests/test_copy_util.py:5]: It works for 3.8 and above 
https://docs.python.org/3/library/unittest.html

File: common/tests/test_aws_service.py (hunk 2 above)
Comment (karle-nishi) [reply common/tests/test_aws_service.py:3]: It works for 3.8 and above
https://docs.python.org/3/library/unittest.html

**Pull request merged**